src/lib/ccf/index.ts

```diff
@@ -12,426 +25,451 @@ import * as GCP_EMBODIED from './gcp-embodied.json';
-              'Min Watts'
-            ] * cpus,
-          maxWatts:
-            this.computeInstanceUsageByArchitecture['gcp'][architecture][
-              'Max Watts'
-            ] * cpus,
-        },
-        maxvCPUs: parseInt(
-          instance['Platform vCPUs (highest vCPU possible)'],
-          10
-        ),
-      } as IComputeInstance;
-    });
-    AZURE_INSTANCES.forEach((instance: KeyValuePair) => {
-      const cpus = parseInt(instance['Instance vCPUs'], 10);
-      let architecture = instance['Microarchitecture'];
-      if (!(architecture in this.computeInstanceUsageByArchitecture['azure'])) {
-        architecture = 'Average';
-      }
-      this.computeInstances['azure'][instance['Virtual Machine']] = {
-        consumption: {
-          idle: 0,
-          tenPercent: 0,
-          fiftyPercent: 0,
-          hundredPercent: 0,
-          minWatts:
-            this.computeInstanceUsageByArchitecture['azure'][architecture][
-              'Min Watts'
-            ] * cpus,
-          maxWatts:
-            this.computeInstanceUsageByArchitecture['azure'][architecture][
-              'Max Watts'
-            ] * cpus,
-        },
-        name: instance['Virtual Machine'],
-        vCPUs: instance['Instance vCPUs'],
-        maxvCPUs: parseInt(
-          instance['Platform vCPUs (highest vCPU possible)'],
-          10
-        ),
-      } as IComputeInstance;
-    });
-    AWS_EMBODIED.forEach((instance: KeyValuePair) => {
-      this.computeInstances['aws'][instance['type']].embodiedEmission =
-        instance['total'];
-    });
-    GCP_EMBODIED.forEach((instance: KeyValuePair) => {
-      this.computeInstances['gcp'][instance['type']].embodiedEmission =
-        instance['total'];
-    });
-    AZURE_EMBODIED.forEach((instance: KeyValuePair) => {
-      this.computeInstances['azure'][instance['type']].embodiedEmission =
-        instance['total'];
-    });
+    this.processInstances(
+      AWS_INSTANCES,
+      'aws',
+      'Instance type',
+      'Platform Total Number of vCPU'
+    );
+    this.processInstances(
+      GCP_INSTANCES,
+      'gcp',
+      'Machine type',
+      'Platform vCPUs (highest vCPU possible)'
+    );
+    this.processInstances(
+      AZURE_INSTANCES,
+      'azure',
+      'Virtual Machine',
+      'Platform vCPUs (highest vCPU possible)'
+    );
+
+    this.processEmbodiedEmissions(AWS_EMBODIED, 'aws');
+    this.processEmbodiedEmissions(GCP_EMBODIED, 'gcp');
+    this.processEmbodiedEmissions(AZURE_EMBODIED, 'azure');
+  }
+
+  /**
+   * Initializes instances.
+   */
+  private initializeComputeInstances() {
+    this.computeInstances['aws'] = {};
+    this.computeInstances['gcp'] = {};
+    this.computeInstances['azure'] = {};
   }
 
+  /**
+   * Calculates average of all instances.
+   */
   private calculateAverage(vendor: string, instanceList: KeyValuePair[]) {
-    let min = 0.0;
-    let max = 0.0;
+    let totalMin = 0.0;
```

Comment:
if you got chance, please try with reduce and see if it works

Reply:
done